onnxscript/optimizer/remove_unused_function.py

```diff
@@ -13,41 +13,49 @@
 logger = logging.getLogger(__name__)
 
 
-class UnusedFunctionRemover(ir.passes.NodeTransformer):
+def _clean_up_unused_functions(model: ir.Model, unused: set[ir.OperatorIdentifier]) -> None:
+    """Removes unused functions from the model."""
+    for op_identifier in unused:
+        del model.functions[op_identifier]
+
+    logger.info("Removed %s unused functions", len(unused))
+    logger.debug("Functions left: %s", list(model.functions))
+    logger.debug("Functions removed: %s", unused)
+
+
+class UnusedFunctionRemover(ir.passes.PassBase):
     def __init__(self):
         super().__init__()
         self.used: set[ir.OperatorIdentifier] = set()
 
-    def _call_function(self, function: ir.Function) -> None:
+    def call(self, model: ir.Model) -> ir.passes.PassResult:
+        for node in ir.traversal.RecursiveGraphIterator(model.graph):
```

Comment:
May be useful to move the initialization of `self.used` here, and clear it at the end of the function?

Reply:
Great point, thanks

Reply:
Done